### ISSUE TITLE
Overwriting index of a collection fails

### ISSUE THREAD

**Issue by anz000** [flamingock/mongock#91]
I am using `Mongock v3.3.2` & `Spring Boot 2.1.4`.

I have an entity 'MyClass' that had an index on a field. I want to modify that index and add unique constraint to that index.

Previously:
```
@Document(collection = "myclass")
public class MyClass {
  ...
  @Indexed
  private String id;
  ..
}
```

New:
```
@Document(collection = "myclass")
public class MyClass {
  ...
  @Indexed(unique = true)
  private String id;
  ..
}
```

Since you have to drop the `index` first and create a new `index` that adjusts to the new `unique` constraint, I wanted to perform a Mongock migration to adjust the indexes. However, the application startup fails before running the migration complaining about the index already existing.

I tried using the standalone setup
```
     @Bean("mongock-standalone")
    public Mongock mongockStandalone() {
        String mongoDatabase = getDatabaseName(); // returns database based on current profile
        MongoClient mongoclient = mongoClient();

        Mongock runner = new MongockBuilder(mongoclient, mongoDatabase, "random.test.rawMigration")
                .setLockQuickConfig()
                .build();

        runner.execute();
        return runner;
    }
```
and attached a @DependsOn on the MyClass entity, but it doesn't work either.

I need the migration completed before Spring Boot starts to evaluate the classes. Any pointers would be much appreciated. Thanks.

**Comment by osantana85** [comment]
Hello @anz000 , you are using an old version of Mongock library. We'd recommend you to upgrade to latest version (currenty 5.3.6), for a better support.

In order to can help you we need some additional information, for example the code of the ChangeUnit (legacy ChangeLog/ChangeSet) you are using to execute that index recreation.

For further information you can take a look to Mongock's [documentation](https://docs.mongock.io). It includes migration guide to upgrade up to version 5.

Hope this helps you.

**Comment by anz000** [comment]
I'll look around if upgrading Mongock is feasible for us at the moment.

Meanwhile, here's the migration file:

```
@Slf4j
@Profile({"default", "alpha", "prod"})
@ChangeLog
public class Id_Unique_Index {

    /**
     * Update id index as Unique
     * @param template the mongo template
     */
    @ChangeSet(id= "update-unique-id", order = "1702334347", author = "dev@myclass.com")
    public void update_id_unique_index(MongoTemplate template) {
        MongoCollection<Document> collection = template.getCollection("myclass");
        String collectionName = collection.getNamespace().getCollectionName();

        // Drop the existing index if it exists
        IndexOperations indexOps = template.indexOps(collectionName);
        if (indexOps.getIndexInfo().stream().anyMatch(indexInfo -> "id".equals(indexInfo.getName()))) {
            indexOps.dropIndex("id");
            System.out.println("Dropped Index on id in MyClass collection.");
        }

        // Create a new index with updated options
        indexOps.ensureIndex(new Index().on("id", Sort.Direction.ASC).unique().named("id"));
        System.out.println("Created new Index on id with unique constraint in MyClass collection.");
    }
}
```

Here's my `Mongock` config:

```
    @Bean("mongock-spring-boot")
    public Mongock mongock() {
        String mongoDatabase = getDatabaseName();
        MongoClient mongoclient = mongoClient();

        // build the client and mongock object; make sure the package name is up-to-date
        return new SpringMongockBuilder(mongoclient, mongoDatabase, "random.test.rawMigration")
                .setLockQuickConfig()
                .build();
    }

```

Previously, it used to be `@Indexed` in the `id` property of MyClass. When I put `@Indexed(unique=true)`, the application startup crashes, complaining about duplicate index. When I remove the annotation `@Indexed(unique=true)` in MyClass, the migration goes through and makes the changes. 

Another pointer is, once I comment out `@Indexed(unique=true)` and run the application. Then stop. Add the `@Indexed(unique=true)` to the class and restart, then the application conforms to the new state. 

So basically, the migration needs to complete before the application starts assessing the MyClass.

Thanks.

**Comment by osantana85** [comment]
Assuming you are working with auto-index-creation enabled (spring-data property), your application will create collections and related indexes at startup (before Mongock's execution take place). If you want to manage index creation with Mongock migrations, you should disable auto-index-creation. Could you try it?

**Comment by anz000** [comment]
It does if I disable the `auto-index-creation`. 

Is it not possible to run the Mongock migration once the MongoDB setup can be accessed and before the application sets up the respective classes? I was hinting at the standalone setup. If it can be done with the standalone setup, I'd still need profile's specific mongoDB uri to perform the migration on.

**Comment by osantana85** [comment]
As you are using standalone  runner, you have the control of Mongock's execution. If Springboot / Springdata is initialized before Mongock's execution take place, it's a problem unrelated to Mongock, it's related to Springboot lifecycle.

A workaround you can try is to move the springdata configuration to a secondary class which depends on Mongock's runner bean, and maybe you can execute it before its initialization takes place. Could you try it? You should use `@DependsOn` in the configuration class and not in the entity.

**Comment by anz000** [comment]
We'll probably upgrade to `5.x` version of `Mongock`, and disable the `auto-index-creation` for `Spring-Data-MongoDB`.

I'll reopen if we need to have `auto-index-creation` enabled.

Thanks.
